app/src/main/java/ch/sdp/vibester/activity/ProfileActivity.kt

```diff
@@ -74,7 +78,46 @@ class ProfileActivity : AppCompatActivity() {
      */
     private fun setEditUserNameBtnListener() {
         findViewById<ImageView>(R.id.editUser).setOnClickListener {
-            showGeneralDialog(R.id.username, "username")
+            showGeneralDialog( "username", true)
+        }
+    }
+    /**
+     * Generic listener for the change profile picture.
+     */
+    private fun setChangeImageBtnListener() {
+        findViewById<ImageView>(R.id.avatar).setOnClickListener {
+            showGeneralDialog( "Do you want to change your profile picture?", false)
+        }
+    }
+    /**
+     * A function that updates the image in the database.
+     * @param id ID of the image int the database
+     */
+
+    private fun updateImage(id: String) {
+        deleteImage(id)
+
+        val intent = Intent(Intent.ACTION_PICK)
+        intent.type = "image/*"
+        startActivityForResult(intent, imageRequestCode)
+    }
+
+
+    /**
+     * A function that deletes an image from the database.
+     * @param id ID of the image int the database
+     */
+    private fun deleteImage(id: String) {
+        imageGetter.deleteImage("profileImg/${id}")
+    }
+
+    //check the UID here not sure
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (resultCode == Activity.RESULT_OK && requestCode == 100){

```

Comment:
I see a 100 there, and there is a field called imageRequestCode which also is 100. If these are supposed to represent the same thing, can you use that one here instead of the static 100 value?